In [21]:
## Extracción de datos


In [22]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [23]:
ruta = "../data/datos_compost.xlsx"

df = pd.read_excel(ruta)

df.head()

,Fecha_Hora,Temp_Invernadero,Humedad_Invernadero,Temp_Exterior,Humedad_Exterior,Temp_Compost,Humedad_Compost
0,2026-05-16 00:08:55,14.1,89.0,14.1,71.0,16.87,89.0
1,2026-05-16 00:07:47,14.3,88.0,14.1,71.0,16.87,88.0
2,2026-05-16 00:06:37,14.4,88.0,14.1,72.0,16.87,88.0
3,2026-05-16 00:05:29,14.7,88.0,14.1,71.0,16.87,88.0
4,2026-05-16 00:04:21,14.2,88.0,14.1,71.0,16.87,88.0


In [24]:
df.columns

Index(['Fecha_Hora', 'Temp_Invernadero', 'Humedad_Invernadero',
       'Temp_Exterior', 'Humedad_Exterior', 'Temp_Compost', 'Humedad_Compost'],
      dtype='str')

In [25]:
# Ejemplo de carga de datos
# df = pd.read_csv('../data/datos.csv')
# df.head()


In [26]:
df.columns

Index(['Fecha_Hora', 'Temp_Invernadero', 'Humedad_Invernadero',
       'Temp_Exterior', 'Humedad_Exterior', 'Temp_Compost', 'Humedad_Compost'],
      dtype='str')

In [27]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2473 entries, 0 to 2472
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Fecha_Hora           2473 non-null   str    
 1   Temp_Invernadero     2473 non-null   float64
 2   Humedad_Invernadero  2473 non-null   float64
 3   Temp_Exterior        2473 non-null   float64
 4   Humedad_Exterior     2473 non-null   float64
 5   Temp_Compost         2473 non-null   float64
 6   Humedad_Compost      2473 non-null   float64
dtypes: float64(6), str(1)
memory usage: 135.4 KB


In [28]:
df["Fecha_Hora"] = pd.to_datetime(df["Fecha_Hora"])

df.head()

,Fecha_Hora,Temp_Invernadero,Humedad_Invernadero,Temp_Exterior,Humedad_Exterior,Temp_Compost,Humedad_Compost
0,2026-05-16 00:08:55,14.1,89.0,14.1,71.0,16.87,89.0
1,2026-05-16 00:07:47,14.3,88.0,14.1,71.0,16.87,88.0
2,2026-05-16 00:06:37,14.4,88.0,14.1,72.0,16.87,88.0
3,2026-05-16 00:05:29,14.7,88.0,14.1,71.0,16.87,88.0
4,2026-05-16 00:04:21,14.2,88.0,14.1,71.0,16.87,88.0


In [29]:
df.isnull().sum()

Fecha_Hora             0
Temp_Invernadero       0
Humedad_Invernadero    0
Temp_Exterior          0
Humedad_Exterior       0
Temp_Compost           0
Humedad_Compost        0
dtype: int64

In [30]:
df.describe()

,Fecha_Hora,Temp_Invernadero,Humedad_Invernadero,Temp_Exterior,Humedad_Exterior,Temp_Compost,Humedad_Compost
count,2473,2473.000000,2473.000000,2473.000000,2473.000000,2473.000000,2473.000000
mean,2026-05-15 01:03:13.833805,20.273393,71.132228,19.123413,72.015366,16.784642,71.132228
min,2026-05-14 01:33:13,11.000000,13.000000,10.500000,25.000000,16.560000,13.000000
25%,2026-05-14 13:15:11,13.600000,38.000000,12.600000,43.000000,16.690000,38.000000
50%,2026-05-15 01:00:46,14.900000,92.000000,14.100000,86.000000,16.750000,92.000000
75%,2026-05-15 12:46:32,27.000000,95.000000,26.200000,95.000000,16.870000,95.000000
max,2026-05-16 00:33:01,40.900000,96.000000,39.500000,98.000000,17.000000,96.000000
std,NaN,8.861428,30.302812,8.635771,25.582244,0.126713,30.302812


In [31]:
def cargar_datos(ruta):
    return pd.read_excel(ruta)


def preparar_datos(df):
    df["Fecha_Hora"] = pd.to_datetime(df["Fecha_Hora"])
    return df


def revisar_datos_faltantes(df):
    return df.isnull().sum()


def resumen_general(df):
    return df.describe()

In [32]:
ruta = "../data/datos_compost.xlsx"

df = cargar_datos(ruta)
df = preparar_datos(df)

df.head()

,Fecha_Hora,Temp_Invernadero,Humedad_Invernadero,Temp_Exterior,Humedad_Exterior,Temp_Compost,Humedad_Compost
0,2026-05-16 00:08:55,14.1,89.0,14.1,71.0,16.87,89.0
1,2026-05-16 00:07:47,14.3,88.0,14.1,71.0,16.87,88.0
2,2026-05-16 00:06:37,14.4,88.0,14.1,72.0,16.87,88.0
3,2026-05-16 00:05:29,14.7,88.0,14.1,71.0,16.87,88.0
4,2026-05-16 00:04:21,14.2,88.0,14.1,71.0,16.87,88.0


In [33]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [34]:
def cargar_datos(ruta):
    return pd.read_excel(ruta)


def preparar_datos(df):
    df["Fecha_Hora"] = pd.to_datetime(df["Fecha_Hora"])
    return df


def revisar_datos_faltantes(df):
    return df.isnull().sum()


def resumen_general(df):
    return df.describe()


ruta = "../data/datos_compost.xlsx"

df = cargar_datos(ruta)
df = preparar_datos(df)

df.head()

,Fecha_Hora,Temp_Invernadero,Humedad_Invernadero,Temp_Exterior,Humedad_Exterior,Temp_Compost,Humedad_Compost
0,2026-05-16 00:08:55,14.1,89.0,14.1,71.0,16.87,89.0
1,2026-05-16 00:07:47,14.3,88.0,14.1,71.0,16.87,88.0
2,2026-05-16 00:06:37,14.4,88.0,14.1,72.0,16.87,88.0
3,2026-05-16 00:05:29,14.7,88.0,14.1,71.0,16.87,88.0
4,2026-05-16 00:04:21,14.2,88.0,14.1,71.0,16.87,88.0


In [35]:
revisar_datos_faltantes(df)

Fecha_Hora             0
Temp_Invernadero       0
Humedad_Invernadero    0
Temp_Exterior          0
Humedad_Exterior       0
Temp_Compost           0
Humedad_Compost        0
dtype: int64

In [36]:
resumen_general(df)

,Fecha_Hora,Temp_Invernadero,Humedad_Invernadero,Temp_Exterior,Humedad_Exterior,Temp_Compost,Humedad_Compost
count,2473,2473.000000,2473.000000,2473.000000,2473.000000,2473.000000,2473.000000
mean,2026-05-15 01:03:13.833805,20.273393,71.132228,19.123413,72.015366,16.784642,71.132228
min,2026-05-14 01:33:13,11.000000,13.000000,10.500000,25.000000,16.560000,13.000000
25%,2026-05-14 13:15:11,13.600000,38.000000,12.600000,43.000000,16.690000,38.000000
50%,2026-05-15 01:00:46,14.900000,92.000000,14.100000,86.000000,16.750000,92.000000
75%,2026-05-15 12:46:32,27.000000,95.000000,26.200000,95.000000,16.870000,95.000000
max,2026-05-16 00:33:01,40.900000,96.000000,39.500000,98.000000,17.000000,96.000000
std,NaN,8.861428,30.302812,8.635771,25.582244,0.126713,30.302812


In [19]:
columnas_numericas = [
    "Temp_Invernadero",
    "Humedad_Invernadero",
    "Temp_Exterior",
    "Humedad_Exterior",
    "Temp_Compost",
    "Humedad_Compost"
]

for columna in columnas_numericas:
    df[columna] = pd.to_numeric(df[columna], errors="coerce")

df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2473 entries, 0 to 2472
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype         
---  ------               --------------  -----         
 0   Fecha_Hora           2473 non-null   datetime64[us]
 1   Temp_Invernadero     2473 non-null   float64       
 2   Humedad_Invernadero  2473 non-null   float64       
 3   Temp_Exterior        2473 non-null   float64       
 4   Humedad_Exterior     2473 non-null   float64       
 5   Temp_Compost         2473 non-null   float64       
 6   Humedad_Compost      2473 non-null   float64       
dtypes: datetime64[us](1), float64(6)
memory usage: 135.4 KB


In [20]:
df[["Temp_Compost", "Humedad_Compost"]].describe()

,Temp_Compost,Humedad_Compost
count,2473.000000,2473.000000
mean,16.784642,71.132228
std,0.126713,30.302812
min,16.560000,13.000000
25%,16.690000,38.000000
50%,16.750000,92.000000
75%,16.870000,95.000000
max,17.000000,96.000000


df.to_csv("../data/datos_compost_limpios.csv", index=False)

print("Archivo limpio guardado correctamente.")

## Conclusión

En esta primera etapa se realizó la carga y preparación inicial de los datos del proceso de compostaje. Se verificó que el archivo contiene 2473 registros y las variables principales de temperatura y humedad del invernadero, ambiente exterior y compost.

También se revisaron los datos faltantes y se evidenció que no existen valores nulos en las columnas analizadas. Además, se convirtió la variable `Fecha_Hora` a formato de fecha, lo que permite ordenar y analizar los registros de manera temporal.

Con esta limpieza inicial, los datos quedan preparados para que los demás integrantes puedan realizar los análisis específicos de temperatura, humedad, comparación ambiental y conclusiones generales del proceso de compostaje.